scripts/pylib/pytest-twister-harness/src/twister_harness/fixtures.py

```diff
@@ -66,3 +68,11 @@ def is_mcumgr_available() -> None:
 @pytest.fixture()
 def mcumgr(is_mcumgr_available: None, dut: DeviceAdapter) -> Generator[MCUmgr, None, None]:
     yield MCUmgr.create_for_serial(dut.device_config.serial)
+
+
+@pytest.fixture(scope='function')
+def gdb_batch(dut: DeviceAdapter, gdb_target_remote: str, gdb_script: str, gdb_timeout) -> CompletedProcess:
```

Comment:
> This fixture cannot be here (in pytst-twister-harness plugin). It uses local fixtures, defined in test dir (options added in conftest.py)

yes, it might be a problem to depend on some external/local fixtures.

@gchwier, what might be a better way for pytest plugin to use command line parameters here ?

Reply:
For now the easiest solution is to keep that fixture in conftest.py in local test dir.
But if we want to have it available for other tests, than we can add command line parameters to [pytest-twister-harness/src/twister_harness/plugin.py](https://github.com/zephyrproject-rtos/zephyr/blob/f6185221440b7fe9f0c1b17702c8231105edc06b/scripts/pylib/pytest-twister-harness/src/twister_harness/plugin.py#L21)
and grab them in a fixture:
```
@pytest.fixture(scope='function')
def gdb_batch(dut: DeviceAdapter, request: pytest.FixtureRequest) -> CompletedProcess:
    gdb_target_remote = request.config.getoption('--gdb-target-remote')
    gdb_script = request.config.getoption('--gdb-script')
    gdb_timeout = request.config.getoption('--gdb-timeout')
    gdb = GDBrunner(dut)
    return gdb.run_script(gdb_target_remote, gdb_script, gdb_timeout)
```
Could be fine also to check, if gdb runner is available etc. and then use `pytest.skip(with reason)`

Reply:
> This fixture cannot be here (in pytst-twister-harness plugin). It uses local fixtures, defined in test dir (options added in conftest.py)

changed to be a factory method fixture

Reply:
> changed to be a factory method fixture

IMHO it is not better solution. Why do you keep that fixture in the plugin, if you need add another fixture in local conftest (to just pass parameters)?
Better to keep `gdb.py` in the plugin helpers methods, and in local conftest add a fixture, that takes given parameters and calls `GDBrunner`.
Or to add parameters to `plugin.py` and keep gdb fixture in the plugin (then please update `pytest.rst` docs, there is a section with available fixtures: https://docs.zephyrproject.org/latest/develop/test/pytest.html#helpers-fixtures )

Reply:
changed back to have gdb fixture local to the test, so no changes at the pytest plugin in this PR.